src/DssSpell.sol

```diff
@@ -50,78 +53,163 @@ contract DssSpellAction is DssAction {
-    uint256 internal constant SIXTEEN_PT_SEVEN_FIVE_PCT_RATE = 1000000004910727769570159235;
-    uint256 internal constant SEVENTEEN_PT_TWO_FIVE_PCT_RATE = 1000000005046239908035965222;
+    uint256 internal constant THIRTEEN_PCT_RATE               = 1000000003875495717943815211;
+    uint256 internal constant THIRTEEN_PT_TWO_FIVE_PCT_RATE   = 1000000003945572635100236468;
+    uint256 internal constant THIRTEEN_PT_SEVEN_FIVE_PCT_RATE = 1000000004085263575156219812;
+    uint256 internal constant FOURTEEN_PCT_RATE               = 1000000004154878953532704765;
+    uint256 internal constant FOURTEEN_PT_TWO_FIVE_PCT_RATE   = 1000000004224341833701283597;
+    uint256 internal constant FOURTEEN_PT_FIVE_PCT_RATE       = 1000000004293652882321576158;
+    uint256 internal constant FOURTEEN_PT_SEVEN_FIVE_PCT_RATE = 1000000004362812761691191350;
+    uint256 internal constant FIFTEEN_PT_TWO_FIVE_PCT_RATE    = 1000000004500681640286189459;
 
     // ---------- Math ----------
-    uint256 internal constant MILLION = 10 ** 6;
-    uint256 internal constant BILLION = 10 ** 9;
-
-    // ---------- Contracts ----------
-    address internal immutable MCD_PAUSE = DssExecLib.getChangelogAddress("MCD_PAUSE");
-
-    // ---------- Trigger Spark Proxy Spell ----------
-    // Spark Proxy: https://github.com/marsfoundation/sparklend/blob/d42587ba36523dcff24a4c827dc29ab71cd0808b/script/output/1/primary-sce-latest.json#L2
-    address internal constant SPARK_PROXY = 0x3300f198988e4C9C63F75dF86De36421f06af8c4;
-    address internal constant SPARK_SPELL = 0xFc72De9b361dB85F1d126De9cac51A1aEe8Ce126;
-
-    function actions() public override {
-        // ---------- DSR Change ----------
-        // Forum: https://forum.makerdao.com/t/accelerated-proposal-rate-system-gsm-delay-psm-usdc-a-ttl-changes/23824
+    uint256 internal constant THOUSAND = 10 ** 3;
+    uint256 internal constant MILLION  = 10 ** 6;
+    uint256 internal constant RAD      = 10 ** 45;
 
-        // Increase the DSR by 10% from 5% to 15%
-        DssExecLib.setDSR(FIFTEEN_PCT_RATE, /* doDrip = */ true);
+    // ---------- Addesses ----------
+    address internal immutable MCD_VOW                 = DssExecLib.vow();
+    address internal immutable MCD_FLAP                = DssExecLib.flap();
 
-        // ---------- Stability Fee Changes ----------
-        // Forum: https://forum.makerdao.com/t/accelerated-proposal-rate-system-gsm-delay-psm-usdc-a-ttl-changes/23824
+    address internal immutable D3M_HUB                 = DssExecLib.getChangelogAddress("DIRECT_HUB");
+    address internal immutable D3M_MOM                 = DssExecLib.getChangelogAddress("DIRECT_MOM");
+    address internal constant MORPHO_D3M_PLAN          = 0x374b5f915aaED790CBdd341E6f406910d648fD39;
+    address internal constant MORPHO_D3M_POOL          = 0x9C259F14E5d9F35A0434cD3C4abbbcaA2f1f7f7E;
+    address internal constant MORPHO_D3M_ORACLE        = 0xA5AA14DEE8c8204e424A55776E53bfff413b02Af;
```

Comment:
This is not needed since `DIRECT-SPARK-MORPHO-DAI` is an ilk and is added to the `DIRECT_HUB`. This whole flow is actually tested in the D3M repository inside [base integration test called `test_unwind_mcd_caged_skimmed`](https://github.com/makerdao/dss-direct-deposit/blob/8a18cb35c19e30f9527f7b1489f79cfcf33a7fa5/src/tests/integration/IntegrationBase.t.sol#L394-L462). Just to be on the safe side, I've added a similar test called `testDirectSparkMorphoCage` (which is a slightly modified version of [the `testRWA015_SPELL_CAGE`](https://github.com/makerdao/spells-mainnet/blob/a4e7963172c6d3691bacb5e487a093e69b16c4fd/archive/2023-06-14-DssSpell/DssSpell.t.sol#L1509-L1562) from 2023-06-14).

In addition to that, you can see that other D3M contracts haven't relied `MCD_ESM` as well, for example:
- Calling [`DIRECT_SPARK_DAI_POOL.wards(0x09e05ff6142f2f9de8b6b65855a1d56b6cfe4c58)`](https://etherscan.io/address/0xafa2dd8a0594b2b24b59de405da9338c4ce23437#readContract#F15) returns 0
- Calling [`DIRECT_SPARK_DAI_PLAN.wards(0x09e05ff6142f2f9de8b6b65855a1d56b6cfe4c58)`](https://etherscan.io/address/0x104fadbb7e17db1a685bba61007dfb015206a4d2#readContract#F6) returns 0
- Calling [`DIRECT_SPARK_DAI_ORACLE.wards(0x09e05ff6142f2f9de8b6b65855a1d56b6cfe4c58)`](https://etherscan.io/address/0xcbd53b683722f82dc82eba7916065532980d4833#readContract#F6) returns 0
- Calling [`DIRECT_COMPV2_DAI_POOL.wards(0x09e05ff6142f2f9de8b6b65855a1d56b6cfe4c58)`](https://etherscan.io/address/0x621fe4fde2617ea8ffade08d0ff5a862ad287ec2#readContract#F14) returns 0